lib/ecto/pool.ex

```diff
@@ -108,7 +114,7 @@ defmodule Ecto.Pool do
   Called once the transaction at `depth` `1` is finished, if the transaction
   is not broken with `break/2`.
   """
-  defcallback close_transaction(t, worker, timeout) :: :ok when worker: any
+  defcallback close_transaction(t, worker, timeout) :: :ok
```

Comment:
No need because checkout_transaction is an optimization that does checkout + open, there is nothing to optimize on the close + checkin.